test/suites/smoke-test-dancelight/test-external-validator-slashes.ts

```diff
@@ -87,5 +87,116 @@ describeSuite({
+
+                // We shouldn't have more bonded eras than the bondedDuration
+                expect(allBondedEras.length).to.be.lessThanOrEqual(bondedDuration.toNumber() + 1);
+
+                // All bondedEras era indexes should be greater or equal than firstKeptIndex
+                expect(
+                    allBondedEras.every((bondedEra) => {
+                        return bondedEra[0].toNumber() >= firstKeptIndex;
+                    })
+                ).toBe(true);
+            },
+        });
+
+        it({
+            id: "C05",
+            title: "A slash message must never exceed the queuedSlashesProcessedPerBlock limit",
+            timeout: 600000,
+            test: async () => {
+                const bondingDuration = api.consts.externalValidatorSlashes.bondingDuration;
+                const maxQueuedSlashesPerBlock = api.consts.externalValidatorSlashes.queuedSlashesProcessedPerBlock;
+
+                const currentEra = (await api.query.externalValidators.activeEra()).unwrap();
+                const currentEraIndex = currentEra.index.toNumber();
+
+                let eraIndexToAnalyze = currentEraIndex;
+                let startBlockNumberOfEra = await getCurrentEraStartBlock(api);
+
+                log(`Analyzing up to ${bondingDuration} eras in the past`);
+
+                /*
+                We move backwards through the eras, up to 'bondingDuration' eras.
+                For each era, we check if there were any unreported slashes at the start of that era.
+                If there were, we then iterate through the blocks within that era looking for
+                a 'SlashesMessageSent' event. If found, we verify that the slash message
+                does not contain more slashes than 'maxQueuedSlashesPerBlock'.
+                */
+                while (eraIndexToAnalyze > currentEraIndex - bondingDuration.toNumber() && eraIndexToAnalyze > 0) {
+                    // Identify the start/end blocks for the previous era
+                    const lastBlockNumberPreviousEra = startBlockNumberOfEra - 1;
+                    const firstBlockNumberPreviousEra = await getPastEraStartBlock(api, lastBlockNumberPreviousEra);
+
+                    // Get the count of unreported slashes at the very start of the previous era
+                    const apiAtStartPreviousEra = await api.at(
+                        await api.rpc.chain.getBlockHash(firstBlockNumberPreviousEra)
+                    );
+                    let unreportedSlashCount = (
+                        await apiAtStartPreviousEra.query.externalValidatorSlashes.unreportedSlashesQueue()
+                    ).length;
+
+                    // If there were unreported slashes at the start of the era, look for slash messages
+                    let blockNumber = firstBlockNumberPreviousEra;
+                    while (unreportedSlashCount > 0 && blockNumber <= firstBlockNumberPreviousEra) {
```

Comment:
What's the worst case amount of blocks checked in this loop? bondingDuration * blocksPerEra? How many hours is that?